### PR TITLE
Icmp and icmpv6 add

### PR DESCRIPTION
**Pull request by shramroy**

### CODE REVIEW

**Review by ajbalogh — APPROVED** [review]

**Review by ajbalogh — CHANGES_REQUESTED** [review]
Please hold off on this PR until the packet header x-pattern refactor is merged.
This PR should align with the new x-pattern extension (see the file changes in PR #73)

**Comment by ashutshkumr** [comment]
@shramroy please refer to https://github.com/open-traffic-generator/models/blob/master/flow/packet-headers/arp.yaml to align with new changes.

**Comment by shramroy** [comment]
> @shramroy please refer to https://github.com/open-traffic-generator/models/blob/master/flow/packet-headers/arp.yaml to align with new changes.

Thanks for sharing the reference. I have submitted the changes.

File: flow/packet-headers/icmp.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+              ICMP subtype
+            format: integer
+            length: 8
+            default: 0
+            features: [count, metric_group]
+        checksum:
+          x-field-pattern:
+            description: >-
+              Internet checksum
+            format: integer
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:39]: format should be checksum

File: flow/packet-headers/icmpv6.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+              ICMPv6 subtype
+            format: integer
+            length: 8
+            default: 0
+            features: [count, metric_group]
+        checksum:
+          x-field-pattern:
+            description: >-
+              Internet checksum
+            format: integer
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:35]: format should be checksum

File: flow/packet-headers/icmpv6.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            length: 16
+        message_body:
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:37]: missing default

File: flow/packet-headers/icmp.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            length: 16
+        rest_of_headers:
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:41]: missing default

File: flow/packet-headers/icmp.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+              parameter_problem: 12
+              timestamp: 13
+              timestamp_reply: 14
+              information_request: 15
+              information_reply: 16
+            description: >-
+              ICMP Type
+            format: integer
+            length: 8
+            default: 4
```
Comment (ajbalogh) [inline flow/packet-headers/icmp.yaml:25]: is this the correct default?

File: flow/packet-headers/icmpv6.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+              time_exceeded: 3
+              parameter_problem: 4
+              echo_reply: 129
+              echo_request: 128
+              multicast_listener_query: 130
+            description: >-
+              ICMPv6 Type
+            format: integer
+            length: 8
+            default: 4
```
Comment (ajbalogh) [inline flow/packet-headers/icmpv6.yaml:21]: is this the correct default?

**Review by ajbalogh — CHANGES_REQUESTED** [review]

File: flow/packet-headers/icmp.yaml (hunk 1 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml]: Updated

File: flow/packet-headers/icmpv6.yaml (hunk 2 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml]: Updated

File: flow/packet-headers/icmpv6.yaml (hunk 3 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml:37]: For Checksum any default value is not added.  I verified with the other protocols as we..

File: flow/packet-headers/icmp.yaml (hunk 4 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml:41]: For Checksum any default value is not added.  I verified with the other protocols as well.

File: flow/packet-headers/icmp.yaml (hunk 5 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml]: Sorry this should be 3 i.e. destination unreachable. I modified it.

File: flow/packet-headers/icmpv6.yaml (hunk 6 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml]: Sorry this should be 1 i.e. destination unreachable. I modified it.

File: flow/packet-headers/icmp.yaml (hunk 4 above)
Comment (ajbalogh) [reply flow/packet-headers/icmp.yaml:41]: the default is missing from the rest_of_headers field

File: flow/packet-headers/icmpv6.yaml (hunk 3 above)
Comment (ajbalogh) [reply flow/packet-headers/icmpv6.yaml:37]: the default is missing from the message_body

File: flow/packet-headers/icmp.yaml (hunk 4 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml:41]: Added Default.

File: flow/packet-headers/icmpv6.yaml (hunk 3 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml:37]: Added default

**Review by ajbalogh — APPROVED** [review]

File: flow/packet-headers/icmp.yaml (hunk 5 above)
Comment (ashutshkumr) [reply flow/packet-headers/icmp.yaml]: I think it should default to `echo_request`

File: flow/packet-headers/icmpv6.yaml (hunk 6 above)
Comment (ashutshkumr) [reply flow/packet-headers/icmpv6.yaml]: I think it should default to `echo_request`

File: flow/packet-headers/icmp.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        checksum:
+          x-field-pattern:
+            description: >-
+              Internet checksum
+            format: checksum
+            length: 16
+        rest_of_headers:
+          x-field-pattern:
+            description: >-
+              ICMP type and code.
```
Comment (ashutshkumr) [inline flow/packet-headers/icmp.yaml:44]: description is incorrect.

Comment (shramroy) [reply flow/packet-headers/icmp.yaml]: Changed

File: flow/packet-headers/icmpv6.yaml (hunk 6 above)
Comment (shramroy) [reply flow/packet-headers/icmpv6.yaml]: Modified it.

File: flow/packet-headers/icmp.yaml (hunk 5 above)
Comment (shramroy) [reply flow/packet-headers/icmp.yaml]: Modified it.

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**